examples/ios/QuickStartSwiftUI/QuickStart.swift

```diff
@@ -295,30 +290,62 @@ struct ItemDetailsView: View {
 
 /// Represents the login screen. We will just have a button to log in anonymously.
 struct LoginView: View {
-    @ObservedObject var auth: Auth
+    @EnvironmentObject var state: AppState
+    // Display an error if it occurs
+    @State var error: Error?
+
     var body: some View {
-        Button("Log in anonymously", action: auth.logIn).disabled(auth.isLoggingInOrOut)
+        VStack {
+            if let error = error {
+                Text("Error: \(error.localizedDescription)")
+            }
+            Button("Log in anonymously") {
+                state.shouldIndicateActivity = true
+                app.login(credentials: .anonymous).receive(on: DispatchQueue.main).sink(receiveCompletion: {
+                    switch ($0) {
+                    case .finished:
+                        state.shouldIndicateActivity = false
+                        break
+                    case .failure(let error):
+                        self.error = error
+                    }
+                }, receiveValue: {
+                    self.error = nil
+                    state.loginPublisher.send($0)
```

Comment:
Not convinced this is actually cleaner than having an auth class specifically responsible for this.